pkg/routeagent_driver/handlers/healthchecker/healthchecker.go

```diff
@@ -0,0 +1,325 @@
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var RouteAgentUpdateInterval = 60 * time.Second
+
+type Config struct {
+	PingInterval         uint
+	MaxPacketLossCount   uint
+	HealthCheckerEnabled bool
+	NewPinger            func(pinger.Config) pinger.Interface
+}
+
+type controller struct {
+	event.HandlerBase
+	sync.Mutex
+	pingers       map[string]pinger.Interface
+	localNodeName string
+	version       string
+	config        *Config
+	stopCh        chan struct{}
+	client        v1typed.RouteAgentInterface
+}
+
+var logger = log.Logger{Logger: logf.Log.WithName("HealthChecker")}
+
+func New(config *Config, client v1typed.RouteAgentInterface, version, nodeName string) event.Handler {
+	controller := &controller{
+		pingers:       map[string]pinger.Interface{},
+		config:        config,
+		version:       version,
+		client:        client,
+		stopCh:        make(chan struct{}),
+		localNodeName: nodeName,
+	}
+
+	return controller
+}
+
+func (h *controller) Stop() error {
+	h.Lock()
+	defer h.Unlock()
+
+	for _, p := range h.pingers {
+		p.Stop()
+	}
+
+	h.pingers = map[string]pinger.Interface{}
+
+	close(h.stopCh)
+
+	err := h.client.Delete(context.TODO(),
+		h.localNodeName, metav1.DeleteOptions{})
+	if err != nil && !apiError.IsNotFound(err) {
+		return errors.Wrapf(err, "Error deleting RouteAgent: %s", h.localNodeName)
+	}
+
+	return nil
+}
+
+func (h *controller) RemoteEndpointCreated(endpoint *submarinerv1.Endpoint) error {
+	if !h.config.HealthCheckerEnabled || h.State().IsOnGateway() {
+		return nil
+	}
+
+	return h.processEndpointCreatedOrUpdated(endpoint)
+}
+
+func (h *controller) RemoteEndpointUpdated(endpoint *submarinerv1.Endpoint) error {
+	if !h.config.HealthCheckerEnabled || h.State().IsOnGateway() {
+		return nil
+	}
+
+	return h.processEndpointCreatedOrUpdated(endpoint)
+}
+
+func (h *controller) processEndpointCreatedOrUpdated(endpoint *submarinerv1.Endpoint) error {
+	logger.Infof("Processing Endpoint: %#v", endpoint)
+
+	if endpoint.Spec.HealthCheckIP == "" || endpoint.Spec.CableName == "" {
+		logger.Infof("HealthCheckIP (%q) and/or CableName (%q) for Endpoint %q empty - will not monitor endpoint health",
+			endpoint.Spec.HealthCheckIP, endpoint.Spec.CableName, endpoint.Name)
+		return nil
+	}
+
+	h.Lock()
+	defer h.Unlock()
+
+	if pingerObject, found := h.pingers[endpoint.Spec.CableName]; found {
+		if pingerObject.GetIP() == endpoint.Spec.HealthCheckIP {
+			return nil
+		}
+
+		logger.Infof("HealthChecker is already running for %q - stopping", endpoint.Name)
+		pingerObject.Stop()
+		delete(h.pingers, endpoint.Spec.CableName)
+	}
+
+	pingerConfig := pinger.Config{
+		IP:                 endpoint.Spec.HealthCheckIP,
+		MaxPacketLossCount: h.config.MaxPacketLossCount,
+	}
+
+	if h.config.PingInterval != 0 {
+		pingerConfig.Interval = time.Second * time.Duration(h.config.PingInterval)
+	}
+
+	if h.config.MaxPacketLossCount != 0 {
+		pingerConfig.MaxPacketLossCount = h.config.MaxPacketLossCount
+	}
```

Comment:
This is redundant since you've already set `MaxPacketLossCount` on L138.

Reply:
Removed it from L138